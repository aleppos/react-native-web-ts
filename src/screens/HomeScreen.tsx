import React, {useState, useEffect} from 'react';
import {StyleSheet, View} from 'react-native';
import {Button, Image, Slider, Text} from 'react-native-elements';

import AudioPlaylistController from '../audio-player/AudioPlaylistController';
import {AudioPlayableItem} from '../audio-player/AudioPlayer';

interface Props {
  environment: any;
  navigation: any;
  renderProps: any;
}

const icon = require('../icons/home.jpg');

const testPlaylist: AudioPlayableItem[] = [
  {
    title: 'test title',
    artwork:
      'https://www.sailer-verlag.de/wp-content/uploads/2019/09/I-Love-English-Junior-Kinderzeitschrift-kategorie.jpg',
    id: 'id02',
    url: require('../audio-player/assets/sample-music.mp3'),
    artist: 'unknown',
    duration: 61,
  },
  {
    title: 'test title',
    artwork:
      'https://www.sailer-verlag.de/wp-content/uploads/2019/09/I-Love-English-Junior-Kinderzeitschrift-kategorie.jpg',
    id: 'id01',
    url: require('../audio-player/assets/08AbenteuerUmMitternacht.mp3'),
    artist: 'unknown',
    duration: 206,
  },
  {
    title: 'test title',
    artwork:
      'https://www.sailer-verlag.de/wp-content/uploads/2019/09/I-Love-English-Junior-Kinderzeitschrift-kategorie.jpg',
    id: 'id03',
    url: require('../audio-player/assets/10TomGehtUnterDiePiraten.mp3'),
    artist: 'unknown',
    duration: 435,
  },
];

export default (props: any) => {
  const {navigation, environment} = props;
  const [position, setPosition] = useState(0);
  const [duration, setDuration] = useState(0);

  useEffect(() => {
    async function reset() {
      AudioPlaylistController.clearPlaylist();
      await AudioPlaylistController.addToPlaylist(...testPlaylist);
      AudioPlaylistController.addTrackChangeListener(trackChanged);
    }
    reset();
  }, []);

  const trackChanged = (track: AudioPlayableItem | null) => {
    if (track) {
      setDuration(track.duration);
      setPosition(0);
    }
  };

  const togglePlay = async () => {
    await AudioPlaylistController.togglePlay();
  };

  const slideTrack = async (position: number) => {
    await AudioPlaylistController.seekTo(position);
  };

  const playPreviousTrack = async () => {
    await AudioPlaylistController.previous();
  };

  const playNextTrack = async () => {
    await AudioPlaylistController.next();
  };

  return (
    <View style={styles.container}>
      <Text>Duration: {duration}</Text>
      <Image
        source={{
          uri:
            'https://www.sailer-verlag.de/wp-content/uploads/2019/09/I-Love-English-Junior-Kinderzeitschrift-kategorie.jpg',
        }}
        style={{width: 200, height: 200}}
      />
      <Slider
        style={styles.slider}
        maximumValue={duration}
        minimumValue={0}
        minimumTrackTintColor="#307ecc"
        maximumTrackTintColor="#000000"
        value={position}
        thumbStyle={styles.thumbstyle}
        onValueChange={value => setPosition(value)}
        onSlidingComplete={slideTrack}
      />
      <Text>Value: {position}</Text>
      <Button
        title="<<"
        onPress={playPreviousTrack}
        style={{width: 200, height: 10}}
      />
      <Button
        title="play"
        onPress={togglePlay}
        style={{width: 200, height: 10}}
      />
      <Button
        title=">>"
        onPress={playNextTrack}
        style={{width: 200, height: 10}}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  thumbstyle: {
    backgroundColor: '#000000',
    height: 10,
  },
  slider: {
    backgroundColor: '#ecf0f1',
    width: 300,
  },
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#fcf0f1',
  },
});
