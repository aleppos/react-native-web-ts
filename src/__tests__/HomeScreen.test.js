//  Generated by relay-snapshot-test-generator
import React from 'react';
import HomeScreen from '../screens/HomeScreen';
import {createMockEnvironment, MockPayloadGenerator} from 'relay-test-utils';
import ReactTestRenderer from 'react-test-renderer';

test('HomeScreen.test Render State', () => {
  const environment = createMockEnvironment();
  const renderer = ReactTestRenderer.create(
    <HomeScreen environment={environment} />,
  );
  environment.mock.resolveMostRecentOperation(operation =>
    MockPayloadGenerator.generate(operation),
  );
  expect(renderer).toMatchSnapshot();
});
