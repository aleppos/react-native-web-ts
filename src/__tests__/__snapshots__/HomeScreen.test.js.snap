// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen.test Render State 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "position": "relative",
      }
    }
  >
    <Image
      onLoad={[Function]}
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 128,
            "width": 128,
          },
        ]
      }
      testID="RNE__Image"
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    />
    <View
      accessibilityElementsHidden={true}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#bdbdbd",
            "height": 128,
            "justifyContent": "center",
            "width": 128,
          }
        }
        testID="RNE__Image__placeholder"
      />
    </View>
    <View
      style={
        Object {
          "height": 128,
          "width": 128,
        }
      }
    />
  </View>
</View>
`;
